examples/generate_image_dataset/generate_images.py

```diff
@@ -114,34 +114,7 @@ def single_step(self, action=None, get_observation=False):
                 render_pass_dir = os.path.realpath(os.path.join(args.images_dir, render_pass))
                 assert os.path.exists(render_pass_dir)
 
-                obs_render_pass = obs["camera." + render_pass]
-                assert len(obs_render_pass.shape) == 3
-                assert obs_render_pass.shape[2] == 4
-
-                if render_pass == "depth":
-                    obs_render_pass_vis = obs_render_pass[:,:,[0,1,2]].copy() # depth is returned as RGBA
-
-                    # discard very large depth values
-                    max_depth_meters = 20.0
-                    obs_render_pass_vis = obs_render_pass_vis[:,:,0]
-                    obs_render_pass_vis = np.clip(obs_render_pass_vis, 0.0, max_depth_meters)
-
-                elif render_pass == "final_color":
-                    obs_render_pass_vis = obs_render_pass[:,:,[2,1,0]].copy() # final_color is returned as BGRA
-
-                elif render_pass == "normal":
-                    obs_render_pass_vis = obs_render_pass[:,:,[0,1,2]].copy() # normal is returned as RGBA
-
-                    # discard normals that aren't properly normalized, i.e., length of 1.0
-                    discard_mask = np.logical_not(np.isclose(np.linalg.norm(obs_render_pass_vis, axis=2), 1.0, rtol=0.001, atol=0.001))
-                    obs_render_pass_vis = np.clip((obs_render_pass_vis + 1.0) / 2.0, 0.0, 1.0)
-                    obs_render_pass_vis[discard_mask] = np.nan
-
-                elif render_pass == "segmentation":
-                    obs_render_pass_vis = obs_render_pass[:,:,[2,1,0]].copy() # segmentation is returned as BGRA
-
-                else:
-                    assert False
+                obs_render_pass_vis = spear.get_image_data(render_pass, obs)
```

Comment:
- I don't want this functionality to be part of `spear`.
- `get_image_data` assumes too much about the structure of the observations, i.e., it assumes there is a single camera and each observation is named `camera.whatever`. What happens when we attach multiple cameras to an agent?
- It is ok with me if this code part of `examples/common`, but it should be generalized.
- It should be in a file called `observation_utils`, it should be called `modify_observation_for_visualization`, and it should be called as follows:

```
observation_components_to_modify = { render_pass: ["camera." + render_pass] for render_pass in config.SIMULATION_CONTROLLER.CAMERA_AGENT.CAMERA.RENDER_PASSES }
obs = observation_utils.modify_observation_for_visualization(obs, observation_components_to_modify)
```

Note that this modified function signature allows multiple depth, color, normal, etc images with arbitrary names, and it allows the user to specify which subset of observations they want to modify.

Reply:
Agreed, updated the code.